pkg/chartrenderer/default.go

```diff
@@ -24,15 +24,15 @@ import (
 	"path/filepath"
 	"strings"
 
+	helmchart "helm.sh/helm/v3/pkg/chart"
+	helmloader "helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/engine"
+	"helm.sh/helm/v3/pkg/releaseutil"
 	"k8s.io/apimachinery/pkg/version"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
-	"k8s.io/helm/pkg/chartutil"
-	"k8s.io/helm/pkg/engine"
 	"k8s.io/helm/pkg/ignore"
```

Comment:
> One solution is to copy https://github.com/helm/helm/blob/main/internal/ignore/rules.go file in gardener.

This does not sound like a sustainable solution. There are many things copied under third_party and I don't see anyone working proactively on getting rid of them.

Reply:
@ialidzhikov  WDYT about https://github.com/gardener/gardener/pull/8877#discussion_r1413944259.

Reply:
It seems that the corresponding pkg in helm2 was public: https://github.com/helm/helm/tree/release-2.17/pkg/ignore
In helm3 it is internal pkg: https://github.com/helm/helm/tree/release-3.13/internal/ignore

I would try to check what was the motivation for this change? Can it be the case that the functionality from the internal pkg is provided by default (w/o explicit configuration) or via other config options?

Reply:
Helm doesn't need to provide an ignore package as it can internally process the `.helmignore` file, but we use the same functionality in our code base for [embed charts](https://github.com/gardener/gardener/blob/0af0b2209fd9c12f7534aef97384e52c8d00c1a3/pkg/chartrenderer/default.go#L200), and helm doesn't support the embed chart.

Reply:
The `ignore` pkg was made internal with https://github.com/helm/helm/pull/6176. In https://github.com/helm/helm/pull/6176#pullrequestreview-273320867 there was a communication that they can make the corresponding pkgs public if there is a request for it. Having a look into https://github.com/helm/helm/blob/v3.10.3/pkg/chart/loader/directory.go I think every `ChartLoader` implementation (that is different from the default one that reads from the filesystem) has to use the `ignore` pkg to make `.helmignore` work. Hence, I will raise PR for it.

I think we can live with the import of "k8s.io/helm/pkg/ignore" for short-term. If we don't have success with making the ignore pkg public, then we can consider maintaining it under third_party or in the same pkg. At the end it is a single file when you exclude the tests.
WDYT?

Reply:
> I think we can live with the import of "k8s.io/helm/pkg/ignore" for short-term. If we don't have success with making the ignore pkg public, then we can consider maintaining it under third_party or in the same pkg. At the end it is a single file when you exclude the tests.

Sure, sounds perfect.

Reply:
> I think every ChartLoader implementation (that is different from the default one that reads from the filesystem) has to use the ignore pkg to make .helmignore work. 

Yes 

> Hence, I will raise PR for it.

Thanks.

Reply:
I created https://github.com/helm/helm/pull/12662. You could add comment somewhere to drop the dependency to `k8s.io/helm/pkg/ignore` if the above PRs gets merged.